$OpenBSD: patch-browser_extensions_tor-launcher_src_components_tl-process_js,v 1.2 2020/06/13 06:26:01 landry Exp $

Let geoip/geoip6 file paths be set by prefs like everything else and let
the new getTorFile() deal with it.

Index: browser/extensions/tor-launcher/src/components/tl-process.js
--- browser/extensions/tor-launcher/src/components/tl-process.js.orig
+++ browser/extensions/tor-launcher/src/components/tl-process.js
@@ -377,6 +377,8 @@ TorProcessService.prototype =
       var torrcFile = TorLauncherUtil.getTorFile("torrc", true);
       var torrcDefaultsFile =
                     TorLauncherUtil.getTorFile("torrc-defaults", false);
+      var geoipFile = TorLauncherUtil.getTorFile("geoip", false);
+      var geoip6File = TorLauncherUtil.getTorFile("geoip6", false);
       var hashedPassword = this.mProtocolSvc.TorGetPassword(true);
       var controlIPCFile = this.mProtocolSvc.TorGetControlIPCFile();
       var controlPort = this.mProtocolSvc.TorGetControlPort();
@@ -404,19 +406,14 @@ TorProcessService.prototype =
         return;
       }
 
-      // The geoip and geoip6 files are in the same directory as torrc-defaults.
-      var geoipFile = torrcDefaultsFile.clone();
-      geoipFile.leafName = "geoip";
 
-      var geoip6File = torrcDefaultsFile.clone();
-      geoip6File.leafName = "geoip6";
-
       var args = [];
       if (torrcDefaultsFile)
       {
         args.push("--defaults-torrc");
         args.push(torrcDefaultsFile.path);
       }
+      args.push("--ignore-missing-torrc");
       args.push("-f");
       args.push(torrcFile.path);
       args.push("DataDirectory");
